### ISSUE TITLE
.remove function errors

### ISSUE THREAD

**Issue by yifanwu** [data-8/datascience#423]
**First: does not update `_num_rows`**

Should count the number of rows again and update here, before returning
https://github.com/data-8/datascience/blob/master/datascience/tables.py#L583

This is a direct problem for applying remove and then a new column from apply

```
t = Table().with_columns(
    'letter', make_array('a', 'b', 'c', 'a', 'b'),
    'count',  make_array(9, 3, 3, 1, 2),
    'points', make_array(1, 2, 2, 10, 1),
)

t.remove(1)

# errors
t['new'] = t.apply(lambda v: v+1, 'count')
```

**second**

Removing index 0 is not working; I understand you can do [0], but still weird. Should change checking condition directly to checking for null, not the default `if ...`.

**Comment by yifanwu** [comment]
Happy to do a PR if the complaints/solutions sounds reasonable!

**Comment by adnanhemani** [comment]
Hi @yifanwu, everything you said was reproducible. I've sent a PR fix in #424 - please let me know if this solves all the issues you've described

**Comment by yifanwu** [comment]
yeah lgtm, though as a general comment, this design (imperatively maintaining some derived state) might encourage this kind of errors in the future. Might be nicer to have some memorization layer, though i'm not familiar with numpy `len`.
